extensions/replication/tasks/CopyLocationTask.js

```diff
@@ -211,91 +228,108 @@ class CopyLocationTask extends BackbeatTask {
+                .then(response => performPutObject(response.Body))
+                .catch(err => {
+                    if (!sourceStreamAborted) {
+                        sourceStreamAborted = true;
+                    }
+                    if (err.$metadata?.httpStatusCode === 404) {
+                        log.error('the source object was not found', Object.assign({
+                            method: 'CopyLocationTask._getAndPutObjectOnce',
+                            peer: this.sourceConfig.s3,
+                            error: err.message,
+                            httpStatus: err.$metadata?.httpStatusCode,
+                        }, actionEntry.getLogInfo()));
+                        return doneOnce(err);
+                    }
                     log.error('an error occurred on getObject from S3',
                               Object.assign({
                                   method: 'CopyLocationTask._getAndPutObjectOnce',
                                   peer: this.sourceConfig.s3,
                                   error: err.message,
-                                  httpStatus: err.statusCode,
+                                  httpStatus: err.$metadata?.httpStatusCode,
                               }, actionEntry.getLogInfo()));
-                }
-                return doneOnce(err);
-            });
-            incomingMsg = sourceReq.createReadStream();
-            incomingMsg.on('error', err => {
-                if (err.statusCode === 404) {
-                    log.error('the source object was not found', Object.assign({
-                        method: 'CopyLocationTask._getAndPutObjectOnce',
-                        peer: this.sourceConfig.s3,
-                        error: err.message,
-                        httpStatus: err.statusCode,
-                    }, actionEntry.getLogInfo()));
-                    return doneOnce(errors.ObjNotFound);
-                }
-                if (!aborted) {
-                    log.error('an error occurred when streaming data from S3',
-                              Object.assign({
-                                  method: 'CopyLocationTask._getAndPutObjectOnce',
-                                  peer: this.sourceConfig.s3,
-                                  error: err.message,
-                              }, actionEntry.getLogInfo()));
-                }
-                return doneOnce(err);
-            });
-            log.debug('putting data', actionEntry.getLogInfo());
+                    return doneOnce(err);
+                });
         }
-        const putDone = err => {
-            if (err && sourceReq) {
-                sourceReq.abort();
-                aborted = true;
-            }
-            doneOnce(err);
-        };
-        return this._sendMultipleBackendPutObject(
-            actionEntry, objMD, size, incomingMsg, log, putDone);
+        return performPutObject(undefined);
```

Comment:
this is a kind of pathologic case, where we can call `_sendMultipleBackendPutObject` directly
without much setup : would be much simpler to handle this case early, so we can see the "nominal"
case more easily, do not need to handle the extra case where incomingMsg is falsish, and avoid the
logic reordering...

```suggestion
if (size == 0) {
    return this._sendMultipleBackendPutObject(actionEntry, objMD, 0, "", log, doneOnce);
}

// continue with the nominal case
```

Reply:
yeah early return is very good idea, dont know why i didnt do this earlier, it also helps to reduce nested code